### DISCUSSION TITLE
اصلاح رابط کاربری تنظیمات

### DISCUSSION THREAD

**Discussion by eshagh79** [omid/Persian-Calendar-for-Gnome-Shell#108] [Ideas]
سلام آقا امید، از اونجایی که صفحهٔ تنظیمات افزونه خیلی درهم و برهم هست، بهتره اون رو بازنویسی کنیم و رابط کاربریش رو از کدهاش جدا کنیم.
رابط کاربریش رو میشه با نرم‌افزار glade به صورت بصری آماده کرد، ولی پیشنهاد من استفاده از libadwaita هست.
اما هنوز glade از اون پشتیبانی نمیکنه.
یه نمونه اولیه از صفحه تنظیمات افزونه که با libadwaita نوشتم رو تصویرش رو میزارم. ببینید چطوره.
۱. هنوز باید یه دکمه منو در سربرگ(هدربار) اضافه بشه، و گزینه‌های «صفحهٔ خانه»، «تماس با ما»، «درباره» به اون اضافه بشه.
۲. یک دکمه «؟» هم باید به سربرگ اضافه بشه، تا «مقادیر ممکن برای قالب‌بندی» در یه پنجرهٔ بازشو رو نمایش بده
۳. بهتره چک‌باکس‌ها هم با سوییچ تعویض بشن، اینطوری قشنگ‌تره
۴. و اینکه پروندهٔ ‎.ui به پروندهٔ جاوااسکریپت تنظیمات متصل بشه
![نماگرفت از 2022-05-29 20-18-00](https://user-images.githubusercontent.com/52137191/170880144-e5419559-fa90-4a0a-8c09-fce6d5ce4daf.png)
میتونید این پرونده رو بعد از تغییرنام(حذف پسوند txt اضافی) با برنامهٔ Workbench مشاهده کنید.
https://workbench.sonny.re
 flatpak install --app flathub re.sonny.Workbench
[new.ui.txt](https://github.com/omid/Persian-Calendar-for-Gnome-Shell/files/8793749/new.ui.txt)

**Comment by eshagh79** [comment]
سلام، یکی از کارها رو انجام دادم، الان چطور به نظر میرسه؟
![نماگرفت از 2022-05-30 14-59-59](https://user-images.githubusercontent.com/52137191/170975436-6b55d021-638a-4bbd-bfe2-102e1afadf01.png)

اینم پرونده ui:
[new.ui.txt](https://github.com/omid/Persian-Calendar-for-Gnome-Shell/files/8797660/new.ui.txt)

**Comment by omid** [comment]
خیلی عالیه.
این ورک‌بنچ رو هم ندیده بودم تا حالا، جالبه :)
چند پیشنهاد دارم.

اول اینکه بالا یا سمت راست (یا چپ بسته به زبان) صفحه tab داشته باشیم و بخش‌های مختلف رو جدا کنیم، مثل «ابزارک سینی»، «رویدادها» و ...
دوم، «علامت سوال» بیشتر برای راهنمای کلی است به نظرم. بهتره این دکمه فقط در کنار فیلدهایی که تاریخ براشون مهمه باشه. (۴-۵ بار تکرار می‌شه، که خوب مهم نیست)

**Reply by eshagh79** [reply]
سلام،
اول اینکه ورک‌بنچ چیه؟
بعد اینکه برای این افزونه از تب استفاده کنیم، منطقی نیست. چون سرجمع سیزده تا قسمت هست، و اگه بخوایم این قسمت‌ها رو گره بندی کنیم و در یک تب جدا قرار بدیم، فقط دسترسی پذیری برنامه رو کمتر کردیم، و باعث میشه کارکردن باهاش سخت‌تر بشه.
حتی در اون شرایط قواعد libadwaita رو هم نقض کردیم، خود این کتابخانه تاکیدش روی اینکه تنظیمات مرتبط به هم در یک صفحه باشه، اگه مثلا در آینده قابلیت اذان یا چیز دیگه به افزونه اضافه شد، اونوقت میشه از یک تب جدا استفاده کرد.
برای موضوع دوم هم خوب مگه مریضیم وقتی میشه با یک دکمه راهنما در سربرگ این مشکل رو حل کرد، سر خودمون رو درد بیاریم و یک دکمه برای هر کدوم از entry ها قرار بدیم؟ این کار هم الان درست نیست.

**Reply by omid** [reply]
همینی که خودت معرفی کردی رو می‌گم دیگه:‌https://workbench.sonny.re/

۱- درست می‌گی، اما دستمون برای آینده و اضافه کردن تنظیمات جدید باز می‌شه. یک لینک به «شرایط قواعد libadwaita» می‌دی در موردش بخونم. ممنون. ضمن اینکه من هم چیز متفاوتی نگفتم. تنظیمات رویدادها، می‌ره توی تب رویدادها!
۲- نه دیگه، اینطوری صحبت کردن درست نیست. این جمله یعنی من مریضم که این حرف رو زدم :) اوکی، به هر حال، من با اینکه همچنان موافق دکمه کنار هر فیلد هستم که هم مشخص‌تر و هدف‌مند‌تر بشه و هم دم دست باشه. اما هر جور راحتی، بذار اون بالا.

**Reply by eshagh79** [reply]
سلام، 
البته «قواعد libadwaita» رو که خالی بستم، ولی مستنداتش اینجاست: https://gnome.pages.gitlab.gnome.org/libadwaita/doc/
اینم کد مبدا: https://gitlab.gnome.org/GNOME/libadwaita
۲- معذرت میخوام که ناراحتتون کردم. یه راهکار دیگه هم این هست که برای هر کدوم از entryها (مدخل‌ها) یه tooltip بزاریم، که وقتی موس روی هردومشون مکث کرد اون راهنما رو نشون بده، فقط ممکنه در تلفن همراه کار نکنه.

**Reply by omid** [reply]
۲- نه بابا، ناراحت نشدم. همینطوری گفتم. آره، هر جوری باشه اوکیه فعلن، بعدن بهترش می‌کنیم.

**Comment by eshagh79** [comment]
آقا امید ببین تا الان تنظیمات افزونه چطور شده؟
[PersianCalendar@oxygenws.com.zip](https://github.com/omid/Persian-Calendar-for-Gnome-Shell/files/8815549/PersianCalendar%40oxygenws.com.zip)

اگه مشکلاتش رو هم فهمیدی به من بگو، چون من که نفهمیدم مشکلش کجاست
